x/evm/types/types.go

```diff
@@ -228,8 +228,6 @@ func (t *ERC20Token) RecordDeployment(txID Hash) error {
 		return fmt.Errorf("token %s non-existent", t.metadata.Asset)
 	case t.Is(Confirmed):
 		return fmt.Errorf("token %s already confirmed", t.metadata.Asset)
-	case t.Is(Pending):
```

Comment:
leave a comment here.

Reply:
what is happening here?

Reply:
we do not need `pending` and `failed` status for token deployment. `non-existence`, `initialize` and `completed` are sufficient to describe its lifecycle. do not want to change it in this pr as it's already large.